### PR TITLE
MaxDegreeOfParallelism defaults for full pruning

### PR DESCRIPTION
**Pull request by MarekM25**
## Changes

We aim to provide default options that meet the needs of most people. 95% of our users want to have full pruning that ideally doesn't affect block processing. The current default `Pruning.FullPruningMaxDegreeOfParallelism = Environment.ProcessorCount` isn't good as it will murder block processing and affects people's attestations. This PR doesn't suppose to be merged as it is but rather an open discussion about defaults.

We have the following options, all of which seem to be better than the current default:
 1. `Pruning.FullPruningMaxDegreeOfParallelism = 2`, 
 2.  `Pruning.FullPruningMaxDegreeOfParallelism = 3`
 3. `Pruning.FullPruningMaxDegreeOfParallelism = 0` but under the hood, it translates to `Environment.ProcessorCount / 2`
 4. `Pruning.FullPruningMaxDegreeOfParallelism = 0` but under the hood, it translates to `Environment.ProcessorCount - 2` 
 5. `Pruning.FullPruningMaxDegreeOfParallelism = 0` but under the hood, it translates to `Environment.ProcessorCount - 1`
 6. Anything else? other constant value than 2 or 3? 
 
 Of course, we will add correct safeguards for Math.Max or Math.Min depends on the option, for example for  `Pruning.FullPruningMaxDegreeOfParallelism = 3` it could be Math.Min(Environment.ProcessorCount, 3)
 
 WDYT @asdacap @LukaszRozmej @kamilchodola @cbermudez97 ? 

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [x] Other: discussion

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Notes on testing

Not now

### CODE REVIEW

**Comment by kamilchodola** [comment]
Option 3 for me looks the best - but remember to take a ceiling value out of it. Better slightly more than slightly less.

**Comment by MarekM25** [comment]
> Option 3 for me looks the best - but remember to take a ceiling value out of it. Better slightly more than slightly less.

Yeah, I think about option 3 or 4. Yes as I described in the issues I am aware of the correct safeguards :)

**Comment by asdacap** [comment]
I don't mind either way. Option 3 is good, but probably not enough. Maybe / 4?. Keep in mind #5588 really throttle it down, by about 2x. #5645 also reduces iops even further. Probably just set to 2 by default, and another option "FullPruningNoThrottle" or something?

**Comment by kamilchodola** [comment]
Just remember about one thing - if we will change defaults and someone decide to use the default, then they will experience quite enormous degradation of pruning time as soon as we will do that. From UX point of view changing for some users from current (maybe) 16 cores down to 3 may be a horrible experience (even if they will have better avg of block processing - ling term it will be the same outcome because it will be slightly worse for longer period of time vs significantly worse but for shorter period).

File: src/Nethermind/Nethermind.Db/IPruningConfig.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public interface IPruningConfig : IConfig
         FullPruningTrigger FullPruningTrigger { get; set; }
 
         [ConfigItem(
-            Description = "'Full' pruning: Defines how many parallel tasks and potentially used threads can be created by full pruning. 0 - number of logical processors, 1 - full pruning will run on single thread. " +
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Db/IPruningConfig.cs:40]: https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.paralleloptions.maxdegreeofparallelism?view=net-7.0 -1 in .NET means Environment.ProcessorCount

**Comment by LukaszRozmej** [comment]
I like the 50% option.

**Comment by benaadams** [comment]
> I like the 50% option.

50% of real cores rather than HT cores?

**Review by kamilchodola — APPROVED** [review]

**Pull request merged**